src/meta/MetaVersionMan.cpp

```diff
@@ -0,0 +1,326 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/MetaVersionMan.h"
+#include "meta/upgradeData/MetaDataUpgrade.h"
+#include "meta/upgradeData/oldThrift/MetaServiceUtilsV1.h"
+#include "meta/processors/jobMan/JobUtils.h"
+#include "meta/processors/jobMan/JobDescription.h"
+
+DEFINE_bool(null_type, true, "set schema to support null type");
+DEFINE_bool(print_info, false, "enable to print the rewrite data");
+
+namespace nebula {
+namespace meta {
+static const int32_t metaVersion = 2;
+static const std::string kMetaVersionKey = "__meta_version__";          // NOLINT
+// static
+int32_t MetaVersionMan::getMetaVersionFromKV(kvstore::KVStore* kv) {
+    CHECK_NOTNULL(kv);
+    std::string value;
+    auto code = kv->get(kDefaultSpaceId, kDefaultPartId, kMetaVersionKey, &value, true);
+    if (code == kvstore::ResultCode::ERR_KEY_NOT_FOUND) {
+        LOG(INFO) << "There is no meta version key existed in kvstore!";
+        if (isV1(kv)) {
+            return 1;
+        } else {
+            auto ret = setMetaVersionToKV(kv);
+            if (!ret) {
+                return -1;
+            }
+            return metaVersion;
+        }
+    } else if (code == kvstore::ResultCode::SUCCEEDED) {
+        if (value.size() != sizeof(int32_t)) {
+            LOG(ERROR) << "Bad format version " << value;
+            return -1;
```

Comment:
Line 39, 44, 49, maybe return 1 will be better? If an invalid version is found, upgrade to 2 sounds reasonable.

Reply:
If reach here, it means the version number is illegal,  the theory should not arrive here. The program needs to exit

Reply:
User will not have a way to modify the version number, then they can not upgrade forever. I think no matter what case, upgrading to 2 is ok, because you have already backup the v1 data.

Reply:
Enter this branch, that has been upgraded, but the format of the version information is wrong, in theory, here should never arrive.